lib/matplotlib/axes/_base.py

```diff
@@ -2673,10 +2673,13 @@ def set_xscale(self, value, **kwargs):
         # around zero
         if value.lower() == 'log' and 'nonposx' not in kwargs.keys():
             kwargs['nonposx'] = 'clip'
-        self.xaxis._set_scale(value, **kwargs)
-        self.autoscale_view(scaley=False)
-        self._update_transScale()
-        self.stale = True
+
+        g = self.get_shared_x_axes()
+        for ax in g.get_siblings(self):
+            ax.xaxis._set_scale(value, **kwargs)
+            ax.autoscale_view(scaley=False)
```

Comment:
Does this need to be called within the loop? I can't quite remember, but doesn't this method check the data limits for all the siblings anyway? We could improve the performance by calling this once for `self` and then explicitly setting the limits for the others, right?